element-connector/src/main/java/org/eclipse/californium/elements/TcpCorrelationContext.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - add support for correlation context to provide
+ *                                      additional information to application layer for
+ *                                      matching messages (fix GitHub issue #1)
+ ******************************************************************************/
+package org.eclipse.californium.elements;
+
+/**
+ * A correlation context that explicitly supports TCP specific properties.
+ */
+public class TcpCorrelationContext extends MapBasedCorrelationContext {
+
+	public static final String KEY_CONNECTION_ID = "CONNECTION_ID";
```

Comment:
I am asking because the term is also used by Thomas Fossati's draft for introducing a *CID* (connection ID) as part of a Hello Extension for DTLS to identify *security associations* independently from the transport. So, why do you want to propagate this Netty property up the layers?